### PR TITLE
timelock: tests the operation store dump

### PR DESCRIPTION
**Pull request by MStreet3**
Tests that dumping the operation store actually writes the expected values to file.  Injects a time faker function and refactors dumpOperationStore to improve testability.

### CODE REVIEW

File: pkg/timelock/scheduler.go [hunk 1]
```diff
@@ -130,7 +132,10 @@ func (tw *Worker) isSchedulerBusy() bool {
 
 // dumpOperationStore dumps to the logger and to the log file the current scheduled unexecuted operations.
 // maps in go don't guarantee order, so that's why we have to find the earliest block.
-func (tw *Worker) dumpOperationStore() {
+//
+// TODO(mstreet3): dumpOperationStore accepts a now func to mock out the time during tests.  Consider
```
Comment (fbac) [inline pkg/timelock/scheduler.go]: Can we remove this TODO comment and create a GH issue out of it?

File: pkg/timelock/scheduler.go [hunk 2]
```diff
@@ -140,35 +145,70 @@ func (tw *Worker) dumpOperationStore() {
...
+		w := bufio.NewWriter(f)
+
+		writeOperationStore(w, tw.logger, tw.store, blocks[0], now)
 
 		w.Flush()
 	}
 }
+
+// writeOperationStore writes the operations to the writer.
+func writeOperationStore(
```
Comment (fbac) [inline pkg/timelock/scheduler.go:161]: nit: Now that you extracted the writeOperationStore I think it makes sense for the timelock-worker store to be independent from the timelock-worker itself, so the tw interacts with it via an interface. This is quite a heavy work, so no need to address it here, I'll create an issue.

**Review by fbac — CHANGES_REQUESTED** [review]

File: pkg/timelock/scheduler.go (hunk 2 above)
Comment (MStreet3) [reply pkg/timelock/scheduler.go:161]: Agreed.. I wanted to stop short of that while increasing some test coverage. I think the TODO above would be handled in such an effort to extract the store.

Comment (MStreet3) [reply pkg/timelock/scheduler.go:161]: https://github.com/smartcontractkit/timelock-worker/issues/34

**Review by fbac — APPROVED** [review]

**Pull request merged**